libs/dal/src/repositories/change/change.schema.ts

```diff
@@ -44,6 +42,13 @@ changeSchema.virtual('user', {
   justOne: true,
 });
 
+changeSchema.index({
+  _parentId: 1,
+  _entityId: 1,
+  _environmentId: 1,
```

Comment:
Long story short let me copy an explanation from the report we got:

● A compound index can be utilized to satisfy multiple queries. For example, if there is
a compound index like { a : 1, b : 1, c:1 }, this can be utilized to satisfy all the
following queries -
* db.coll.find({ a : 3 })
* db.coll.find({ a : 3, b : 5})
* db.coll.find({ a : 3, b : 5, c : 8})

Let me know if it helps to understand the logic behind it better. @p-fernandez

Reply:
I missed an `all together` in my question. So MongoDB does the magic behind scenes. Then good for me. 🤐